### DISCUSSION TITLE
MariaDB - working correctly?

### DISCUSSION THREAD

**Discussion by Loader23** [dani-garcia/vaultwarden#2279] [Q&A]
Hi,

how do I know, that the MariaDB is working correctly? I did not create a database and user and the commands did not work anyway.
But the container created it on it own. I guess the commands are for mysql? 

Anyway: When I create a new entry there is no change to the files inside the data folder. The only file that changes is the ib_logfile0.
When I compose down the files also changes in the vaultwarden db folder. How do I know all changes are always safe?

**Comment by BlackDex** [comment]
If you didn't create anything i guess it is still using SQLite instead of MySQL/MariaDB.
Checkout: https://github.com/dani-garcia/vaultwarden/wiki/Using-the-MariaDB-%28MySQL%29-Backend

**Comment by Loader23** [comment]
No, there is no sqlite db file in the vaultwarden data folder. And when I stop the MariaDB container nothing is shown anymore.

**Comment by BlackDex** [comment]
Then to answer the question it self.
If you have configured the DATABASE_URL correctly and you created a user, then you should see that user in the users table.

**Comment by Loader23** [comment]
The wiki lacks some crucial Information. Why is it not written that you have to use mysql -uroot -p$password before you can use any other command? Or "USE `database_name`;" before you can use the alter commands.

Anyway, I could not create a database because it already existed. I could create a vaultwarden@localhost user but he already had created a vaultwarden@% user at the first container startup so I guess its using that user.

The confusing part is, that the date of the files only change on a shutdown. Like MariaDB is keeping everything in cache or ram before writing it.

**Reply by BlackDex** [reply]
The Vaultwarden repo is not the place to ask specific MySQL/MariaDB questions, that includes how you manage the MariaDB Server it self.

Some people use PHPMyAdmin, some use other tools, or the CLI, and we kinda assume that someone does know how to operate a MariaDB server, that is the same for a PostgreSQL server, this place is not a helpdesk on how to use those services it self.

It has an example there to use a docker container, which does it all for you.
There are also queries there that help you to create the base of the database which you indeed need to execute as a use which has the privilege to do so, that can be root, or any other user which has the rights to do so.

**Reply by Loader23** [reply]
Usually a wiki or a documentation describes the whole process and does not leave parts out because its assumed the user knows everything else. Never had this before, thats all. It was just a suggestion to improve the wiki for new users. :-)

Everything is running fine, with the compose example the user and database is automatically created. So the queries are a bit misleading. Maybe the collation and charset is needed, did not check that.

All I want to know is, does the files inside the database folder change when something is added, deleted or modified inside the Vault. I don't think thats a specific MariaDB question. Everyone who uses Vaultwarden with MariaDB can answer this.

**Reply by BlackDex** [reply]
It's a wiki and you are free to edit this if it is needed.
Though, as i said before, it is not always the root user, and if you have no clue what you are doing, i suggest to use the SQLite instead which solves having to setup a different service.

Regarding the question about the files, that really is a MySQL/MariaDB question, Vaultwarden only talks to the database. And it is a specific service question which has nothing specifically to do with Vaultwarden it self. The people who develop MySQL or MariaDB probably know the answer better.

**Comment by Loader23** [comment]
Good idea with the wiki. I added a line to make it more clear that some steps are not needed. Together with your edits this will help new users.

I just want to know if other Vaultwarden users observe the same. So I still think its the right place to ask that question. Thats why I made it a question and not an Issue. Everything is running fine, just want to know if its normal. It seems you are not using MariaDB so I'll wait for someone who could quickly check the file dates in their mariadb data folder.
